crates/block-producer/src/server/mod.rs

```
@@ -125,38 +124,33 @@ impl BlockProducer {
                 BlockProducerRpcServer::new(mempool, store)
                     .serve(rpc_listener)
                     .await
-                    .expect("Really the rest should throw errors instead of panic'ing.")
+                    .expect("block-producer failed")
             })
             .id();
 
-        // Wait for any task to end. They should run forever, so this is an unexpected result.
+        let task_ids = HashMap::from([
+            (batch_builder_id, "batch-builder"),
+            (block_builder_id, "block-builder"),
+            (rpc_id, "rpc"),
+        ]);
 
+        // Wait for any task to end. They should run indefinitely, so this is an unexpected result.
+        //
         // SAFETY: The JoinSet is definitely not empty.
         let task_result = tasks.join_next_with_id().await.unwrap();
+
         let task_id = match &task_result {
-            Ok((id, _)) => *id,
+            Ok((id, ())) => *id,
             Err(err) => err.id(),
         };
+        let task = task_ids.get(&task_id).unwrap_or(&"unknown");
```

Same comment about panicking instead of defaulting to "unknown"